scene/3d/physics_body.cpp

```diff
@@ -962,8 +962,12 @@ Ref<KinematicCollision> KinematicBody::_move(const Vector3 &p_motion, bool p_inf
 
 	bool collided = move_and_collide(p_motion, p_infinite_inertia, col, p_exclude_raycast_shapes, p_test_only);
 
+	// Ugly hack as a hot fix, 65b3200 fix an issue but cause a problem with Bullet that broke games using Bullet.
+	// The bug is something internal to Bullet, seems to be related to the Bullet’s margin. As not proper fix was found yet,
+	// this temporary solution solves the issue for games using Bullet.
+	bool is_bullet = ProjectSettings::get_singleton()->get(PhysicsServerManager::setting_property_name) != "GodotPhysics";
 	// Don't report collision when the whole motion is done.
```

Comment:
ProjectSettings should not be `got` frequently, they should be cached (perhaps as a static within the physics?).

See `project_settings.h` for details on how to keep this updated in the case of changes, if this can change after startup.

Reply:
I don't think the physics engine can change at runtime, so caching this in a constructor should be enough.

Reply:
I found a cleaner solution without using the `ProjectSettings`, by caching the server ID in PhysicsServerManager (as you suggested).